packages/client/lib/__tests__/CredentialRequestClient.spec.ts

```diff
@@ -138,9 +145,10 @@ describe('Credential Request Client with Walt.id ', () => {
     nock.cleanAll();
     const WALT_IRR_URI =
       'openid-initiate-issuance://?issuer=https%3A%2F%2Fjff.walt.id%2Fissuer-api%2Foidc%2F&credential_type=OpenBadgeCredential&pre-authorized_code=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJhOTUyZjUxNi1jYWVmLTQ4YjMtODIxYy00OTRkYzgyNjljZjAiLCJwcmUtYXV0aG9yaXplZCI6dHJ1ZX0.YE5DlalcLC2ChGEg47CQDaN1gTxbaQqSclIVqsSAUHE&user_pin_required=false';
-    const initiation = IssuanceInitiation.fromURI(WALT_IRR_URI);
+    const initiation: IssuanceInitiationWithBaseUrl = new IssuanceInitiation(WALT_IRR_URI).issuanceInitiationWithBaseUrl;
```

Comment:
Why is something which nicely used a builder like pattern now moved to a constructor?

Reply:
> Use of static on this largge scale limits the ability have multiple instances of the same classes.

Please explain, as they are used to instantiate actual new objects, meaning you can extend and/or add others in the future. 

Typically you avoid statics for OO extensibility. Except when you have builders or want a bit more control on how to instantiate an object. That is what we are doing here, so I fail to see how "static on this large scale limits to have multiple instances of the same class". The static instantiates a new object based on the params!

Reply:
This example maps on to Abstract Factory ... Clients being factories and Initiation and Offer being products. Factories do not require static functions. 
The inaccuracy of implementation is that the there are multiple factory hirarcies ... one hirarchy is the CredentialOfferClient, CredentialIssuanceClient, IssuanceInitiationClient ... and the other is 'MetadataClient' ... 

With the latest changes this metadataClient class has getServerMetaDataFromInitiation & getServerMetaDataFromCredentialOffer. I believe MetadataClient should be converted to the factory hirarchy. 

CredentialOfferMetadataClient
MetadataClient
IssuanceInitiationMetadataClient

OpenID4VCIClient is the client class for our abstract factory pattern. But this directly engages with MetadataClient this should be moved to CredentialOfferClient ...

These were the factors which lead to creating an early PR. And I hope we still need to discuss these points.